src/Bicep.Core/Emit/IntegerValidatorVisitor.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+
+namespace Bicep.Core.Emit
+{
+    public class IntegerValidatorVisitor : SyntaxVisitor
+    {
+        private readonly SemanticModel semanticModel;
```

Comment:
[nit] Doesn't look like you're actually using `semanticModel` in this class - can remove it